### PR TITLE
Add the support of testing on multiple datasets after training on one dataset.

### PR DESCRIPTION
**Pull request by Harry-zzh**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/frameworks/multimodal/exec.py [hunk 1]
```diff
@@ -193,6 +193,7 @@ def run(
         "problem_type": train_data.problem_type,
         "presets": params.pop("presets", None),
         "path": os.path.join(benchmark_dir, "models"),
+        "eval_metric": val_data.metric,
```
Comment (suzhoum) [inline src/autogluon/bench/frameworks/multimodal/exec.py:196]: `val_data` might not be available for all datasets so we need to handle that.

Comment (Harry-zzh) [reply src/autogluon/bench/frameworks/multimodal/exec.py:196]: How about add these lines:

```Python
if val_data is not None:
    predictor_args["eval_metric"] = val_data.metric
````

Comment (suzhoum) [reply src/autogluon/bench/frameworks/multimodal/exec.py:196]: Yes that would work.

Comment (Harry-zzh) [reply src/autogluon/bench/frameworks/multimodal/exec.py]: Done.

**Review by suzhoum — APPROVED** [review]

**Pull request merged**